lib/src/webdav/file.dart

```diff
@@ -1,48 +1,184 @@
 import 'package:intl/intl.dart';
+import 'package:nextcloud/nextcloud.dart';
 import 'package:xml/xml.dart' as xml;
 
 /// WebDavFile class
 class WebDavFile {
-  // ignore: public_member_api_docs
-  WebDavFile(
-    this.path,
-    this.mimeType,
-    this.size,
-    this.lastModified, {
-    this.shareTypes = const [],
-  });
+  /// Creates a new WebDavFile object with the given path
+  WebDavFile(this.path);
 
   // ignore: public_member_api_docs
   final String path;
 
-  // ignore: public_member_api_docs
-  final String mimeType;
+  /// The fileid namespaced by the instance id, globally unique
+  String id;
+
+  /// The unique id for the file within the instance
+  String fileId;
+
+  /// Whether this is a collection resource type
+  bool isCollection = false;
 
   // ignore: public_member_api_docs
-  final int size;
+  String mimeType;
+
+  /// File content length or folder size
+  int size;
+
+  /// The user id of the owner of a shared file
+  String ownerId;
+
+  /// The display name of the owner of a shared file
+  String ownerDisplay;
+
+  /// Share note
+  String note;
 
   // ignore: public_member_api_docs
-  final DateTime lastModified;
+  DateTime lastModified;
+
+  /// Upload date of the file.
+  DateTime uploadedDate;
+
+  /// Creation date of the file as provided by uploader.
+  DateTime createdDate;
 
   // ignore: public_member_api_docs
-  final List<int> shareTypes;
+  List<int> shareTypes = [];
+
+  /// User IDs of sharees.
+  List<String> sharees = [];
+
+  /// Whether this file is marked as favorite.
+  bool favorite = false;
```

Comment:
Is there a reason why these fields can't be final?

Reply:
I'm building the `WebDavFile` iteratively from XML and didn't want to introduce an intermediate builder object.
Given that many fields are optional it also allows for a more intuitive way of defining defaults.

Is there a particular reason you're wanting to keep all of this final?